File: 005.rb
# 2520 is the smallest number that can be
# divided by each of the numbers from 1 to 10
# without any remainder.

# What is the smallest number that is evenly
# divisible by all of the numbers from 1 to 20?

require 'mathn'

max = 20
min = (max / 2).floor + 1
n   = min

(min..max).each do |i|
  n = n.lcm i
end

puts n

File: go/src/euler/005.go
// 2520 is the smallest number that can be divided by each of the numbers from 1 to 10
// without any remainder.

// What is the smallest number that is evenly divisible by all of the numbers from 1 to 20?
package main

import (
	"fmt"
	"euler/lib"
)

func main() {
	var n,i uint64 = 2520,11
	for i = 11; i <= 20; i++ {
		n = lib.Lcm(n,i)
	}
	fmt.Println(n)
}
